internal/versions/versions.go

```diff
@@ -175,42 +171,88 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, osImage := range h.osImages {
+	// Find OS images of relevant CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		// Empty implies default CPU architecture
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == "" || cpuArchitecture == common.DefaultCPUArchitecture
+		}
 		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
 			cpuArchitecture = common.DefaultCPUArchitecture
 		}
-		if *osImage.OpenshiftVersion == versionKey && *osImage.CPUArchitecture == cpuArchitecture {
-			return osImage, nil
-		}
+		return swag.StringValue(osImage.CPUArchitecture) == cpuArchitecture
+	})
+	if funk.IsEmpty(osImages) {
+		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in OS images list", cpuArchitecture)
 	}
 
-	return nil, errors.Errorf("The requested OS image for version (%s) isn't specified in OS images list", versionKey)
-}
+	// Search for specified x.y.z openshift version
+	osImage := funk.Find(osImages, func(osImage *models.OsImage) bool {
+		return *osImage.OpenshiftVersion == openshiftVersion
+	})
 
-// Returns the ReleaseImage entity
-func (h *handler) GetReleaseImage(openshiftVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
 	versionKey, err := h.getKey(openshiftVersion)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, releaseImage := range h.releaseImages {
-		if cpuArchitecture == "" {
+	if osImage == nil {
+		// Fallback to x.y version
+		osImage = funk.Find(h.osImages, func(osImage *models.OsImage) bool {
+			return *osImage.OpenshiftVersion == versionKey
+		})
+	}
+
+	if osImage == nil {
+		// Try to find a patch version by x.y version
+		osImage = funk.Find(h.osImages, func(osImage *models.OsImage) bool {
+			imageVersionKey, err := h.getKey(*osImage.OpenshiftVersion)
+			if err != nil {
+				return false
+			}
+			return imageVersionKey == versionKey
+		})
+	}
+
+	if osImage != nil {
+		return osImage.(*models.OsImage), nil
+	}
+
+	return nil, errors.Errorf("The requested OS image for version (%s) isn't specified in OS images list", openshiftVersion)
+}
+
+// Returns the ReleaseImage entity
+func (h *handler) GetReleaseImage(openshiftVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
+	// Find OS images of relevant CPU architecture
+	releaseImages := funk.Filter(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		if swag.StringValue(releaseImage.CPUArchitecture) == "" {
 			// Empty implies default CPU architecture
-			cpuArchitecture = common.DefaultCPUArchitecture
+			return cpuArchitecture == "" || cpuArchitecture == common.DefaultCPUArchitecture
```

Comment:
Same as before, I think this would be more clear if we set `cpuArchitecture` to the default outside of the filter function.

Reply:
yeah, will handle it the same